google-home-community.groovy

```diff
@@ -4233,11 +4265,15 @@ private traitFromSettings_SensorState(traitName) {
             descriptiveAttribute:           [:],
             numericAttribute:               [:],
             numericUnits:                   [:],
+            reportsDescriptiveState:        [:],
+            reportsNumericState:            [:],
         ]
         sensorMapping.descriptiveState = settings."${traitName}.sensorTypes.${sensorType}.availableStates"
         sensorMapping.descriptiveAttribute = settings."${traitName}.sensorTypes.${sensorType}.descriptiveAttribute"
         sensorMapping.numericAttribute = settings."${traitName}.sensorTypes.${sensorType}.numericAttribute"
         sensorMapping.numericUnits = settings."${traitName}.sensorTypes.${sensorType}.numericUnits"
+        sensorMapping.reportsDescriptiveState = settings."${traitName}.sensorTypes.${sensorType}.reportsDescriptiveState"
+        sensorMapping.reportsNumericState = settings."${traitName}.sensorTypes.${sensorType}.reportsNumericState"
```

Comment:
Since these are all set unconditionally it seems like it would be better to just initialize them straight away instead of starting with a default (which doesn't make sense) and then setting them later:
```
def sensorMapping = [
    descriptiveState:        settings."${traitName}.sensorTypes.${sensorType}.availableStates",
    descriptiveAttribute:    settings."${traitName}.sensorTypes.${sensorType}.descriptiveAttribute",
    numericAttribute:        settings."${traitName}.sensorTypes.${sensorType}.numericAttribute",
    numericUnits:            settings."${traitName}.sensorTypes.${sensorType}.numericUnits",
    reportsDescriptiveState: settings."${traitName}.sensorTypes.${sensorType}.reportsDescriptiveState",
    reportsNumericState:     settings."${traitName}.sensorTypes.${sensorType}.reportsNumericState",
]
```

Reply:
Way cleaner.  Fixed